gc3libs/backends/transport.py

```diff
@@ -473,6 +474,7 @@ def __init__(self, remote_frontend,
         # init connection params
         self.username = None
         self.keyfile = None
+        self.pkey = None
```

Comment:
Shouldn't this be?
```suggestion
        self.pkey = pkey
```
Otherwise the constructor parameter is being ignored...

Reply:
This is done by the `set_connection_params` call:

https://github.com/gc3pie/gc3pie/blob/73e34328fdefafa0a924d10748387b480abeee8e/gc3libs/backends/transport.py#L574

If you'd prefer this to be done in the constructor, I'm happy to change this.

Reply:
I find it a bit confusing that there is a constructor argument `pkey`, the constructor code sets `self.pkey` to `None`, then passes the argument to `set_connection_params()` which sets `self.pkey` to ctor's `pkey`... Has no clear intent, IMHO.

`set_connection_params()` is there to provide a mechanism to merge SSH config files with GC3Pie config -- I don't think this applies to `pkey`.

In short: yes, please set it in the constructor and not in `set_connection_params()`

Reply:
Moved in 58dd19068874b37034a2ce6418185d29a4670b89